src/server/pkg/collection/collection.go

```diff
@@ -568,10 +568,6 @@ func (c *readonlyCollection) Watch() (watch.Watcher, error) {
 	return watch.NewWatcher(c.ctx, c.etcdClient, c.prefix, c.prefix, c.template)
 }
 
-func (c *readonlyCollection) WatchWithPrev() (watch.Watcher, error) {
-	return watch.NewWatcherWithPrev(c.ctx, c.etcdClient, c.prefix, c.prefix, c.template)
-}
-
```

Comment:
As much as I love seeing code being deleted. This seems like a reasonable method to have, even if it's no longer being used.

Reply:
TBH, I feel compelled to push on this just a little bit. I think `WithPrev` is generally unsafe, and part of what made the previous version of the PPS master brittle was that you couldn't rely on prevKey being set—if the master restarted, `prevValue` wasn't set for any of the values returned by the watch, which meant that all of the code that handled watch updates had to query kubernetes anyway to see what was there and deal with unexpected cases.

I think that anywhere we use `withPrev` in the future is going to have this problem—I think it's not a generally safe assumption that a watch restarting is a corner case, and as soon as it happens any logic based on the previous key becomes invalid.

If you still disagree I'm happy to leave it

Reply:
This makes sense, I'm cool with removing it.